inference-engine/ie_bridges/c/include/c_api/ie_c_api.h

```diff
@@ -302,6 +303,47 @@ typedef struct ie_available_devices {
     size_t num_devices;
 }ie_available_devices_t;
 
+/**
+ * @enum shared_context_type_e
+ * @brief Represents the lists of type for supported context.
+ */
+typedef enum {
+    OCL = 0,     // OpenCL context.
+    VA_SHARED,   // VA_Display on Linux and ID3D11Device on Windows.
+}shared_context_type_e;
+
+/**
+ * @struct shared_context_param_t
+ * @brief The structure contains context handle user-supplied and context type paramters
+ * for creating shared context.
+ */
+typedef struct shared_context_param {
+    void *ctx;  // context handle user-supplied, such as cl_context, VADisply, ID3D11Device object handle.
+    shared_context_type_e type;
+}shared_context_param_t;
+
+/**
+ * @enum shared_blob_type_e
+ * @brief Represents the lists of type for supported shared blob.
+ */
+typedef enum {
+    OCL_BUFFER = 0,   // shared OpenCL buffer blob
+    OCL_IMAGE2D,      // shared OpenCL 2D image blob
+    VA_SURFACE,       // shared video decoder surface or D3D 2D texture blob
+    DX_BUFFER,        // shared D3D buffer blob
+}shared_blob_type_e;
+
+/**
+ * @struct shared_blob_param_t
+ * @brief The structure contains shared memory blob handle user-supplied and shared blob type paramters
+ * for creating shared blob.
+ */
+typedef struct shared_blob_param {
+    void *blob_handle;  // memory handle user-supplied, such as cl_mem, VASurfaceID, ID3D11Buffer, ID3D11Texture2D,
+    shared_blob_type_e type;
+    uint32_t plane;
+}shared_blob_param_t;
```

Comment:
All this structs/enums looks like GPU plugin specific.

Reply:
I would expect something like this:

```
struct param {
    int ID;  // or const char* name
    void* value;
};

ie_core_create_context(..., const param* params, int numParams);
```

The C API wrapper should convert the `params` C array to C++ `ParamMap` based on `ID`.

Reply:
Or even introduce separate C API for `ParamMap`:

```
struct param_map_t;
param_map_t* create_param_map();
void free_param_map(param_map_t* params);
void add_int_param(param_map_t* params, const char* name, int value);
void add_float_param(param_map_t* params, const char* name, float value);
...
```